src/QuickActions.ts

```diff
@@ -73,64 +61,95 @@ export class QuickActions {
-        if(priority === Priority.Word){
-            results.word.push(shortcut)
-        }else if(priority === Priority.Tag){
-            results.tag.push(shortcut)
-        }else if(priority === Priority.Start){
-            results.start.push(shortcut)
-        }else if(priority === Priority.TagStart){
-            results.tagStart.push(shortcut)
+    static checkMatch(shortcut: KeyboardShortcut, searchTerm: string): number {
+        const perfectWordMatchPoints = 10000 //search term and shortcut name are a perfect match
+        const startsWithPoints = 100 //shortcut name starts with the search term
+        const wordMatchPoints = 50 //a word in the search term matches a word in the shortcut name
+        const wordMatchInSequencePoints = 2 //extra points each word that matches between shortcut name and search term
+        const partialWordMatchPoints = 8 //some of a word in the search term matches a word of the shortcut name
+        const perfectTagMatchPoints = 40 //search term and tag are a perfect match
+        const tagMatchPoints = 30 //a word in the search term matches a tag
+        const partialTagMatchPoints = 5 //some of a word in the search term matches a tag
+
+        let shortcutMatchScore = 0 //keeping track of this keyboard shortcut's score
+
+        //perfect match! no point to continue
+        if (shortcut.text.toLocaleLowerCase() === searchTerm.toLocaleLowerCase()){
+            return perfectWordMatchPoints
         }
-    }
 
-    static checkMatch(shortcut: KeyboardShortcut, searchTerm: string): Priority | null {
-        let match = false
-
-        // check for match against shortcut text
-        if(shortcut.text.toLocaleLowerCase().includes(searchTerm)){
-            match = true
+        //check if function name starts with the search term
+        if (shortcut.text.toLocaleLowerCase().startsWith(searchTerm.toLocaleLowerCase())){
+            shortcutMatchScore+=startsWithPoints
         }
 
-        // check for match against shortcut tags
-        shortcut.tags.forEach(function(tag){
-            if(tag.toLocaleLowerCase().includes(searchTerm)){
-                match = true
+        for(const tag of shortcut.tags){
+            // check for perfect tag matchs
+            if (tag.toLocaleLowerCase() === searchTerm.toLocaleLowerCase()){
+                shortcutMatchScore += perfectTagMatchPoints
             }
-        })
-
-        // abort if no match found
-        if (!match){
-            return null;
         }
 
+
         // split both the shortcut text and search term into individual words
         const shortcutWords: string[] = shortcut.text.split(' ');
         const searchWords: string[] = searchTerm.split(' ').filter(Boolean);
 
-        for (const searchWord of searchWords){
-            
+        let searchWordStillMatches = true // keeping track of how if the words from the search word match the word keyboardshortcut name of the same index
```

Comment:
Very minor. But I'd use the capitalisation "KeyboardShortcut" in the comment here. To be clear that you are referring to the class/object.

Maybe re-write the comment a little too, it seems to have a couple of extra words? or is a little difficult to understand somehow